httpx/middleware/query_normalizer.go

```diff
@@ -0,0 +1,46 @@
+package middleware
+
+import (
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+// WithQueryNormalizer fixes wrong php query string handling as array
+func WithQueryNormalizer(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if i := strings.Index(r.URL.RawQuery, "["); i == -1 {
+			return
```

Comment:
You need to delegate the request before returning

Reply:
ofc.